### ISSUE TITLE
Alt Shortcuts Don't Work Properly With Top Activity Bar

### ISSUE THREAD

**Issue by arookas** [microsoft/vscode#201613]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.85.1
- OS Version: Windows_NT x64 10.0.19045

Steps to Reproduce:

1. Set `workbench.activityBar.location` to `top`
2. Hide menu bar (View > Appearance > Menu Bar)
3. Hide sidebar with Alt+B
4. Press Alt+F to open file menu

The menu will not appear. When pressing Alt+B afterwards to reveal the sidebar, the menu is entirely glitched:
![image](https://github.com/microsoft/vscode/assets/11038698/8e65e85e-40de-452b-8be8-0a89b66a9108)

A workaround is to Alt+B first, then Alt+F to open the menu. Then it appears correctly:
![image](https://github.com/microsoft/vscode/assets/11038698/0104b77b-b899-411f-9ab0-dade9ec41a40)

**Comment by sbatten** [comment]
@sandy081 I'm not sure if we should have the compact menu in the activity bar when it is on top. Was this already discussed?

**Comment by sandy081** [comment]
> Was this already discussed?
No

I think the right fix here is to open the sidebar when alt+f is pressed

**Comment by sbatten** [comment]
It's not easy to see from the image because the sidebar is obscured by the menu, but the side bar is already open. But I find the application menu being nested in the sidebar quite strange.

**Comment by sandy081** [comment]
@arookas Wondering how the same behaves when activity bar is on the side but hidden?

**Comment by sbatten** [comment]
@sandy081 in that case, alt+f does not trigger the menu, we don't toggle the activity bar in

**Comment by sbatten** [comment]
Those are 2 separate cases:

Case 1: Compact menu lives in the activity bar with activity bar on the side but hidden
Behavior: Alt+F does nothing because the hamburger is hidden with the activity bar

Case 2: Compact menu lives in the activity bar with activity bar on top next to view container icons
Behavior: Alt+F works but the menu is misplaced. 

Case 2 is a bug. If we think that's where the menu should live when the activity bar is on top and the menu is compact, then I will fix the positioning, but I am arguing that the menu should live somewhere else, potentially also back to the title bar like the accounts and settings icon.

**Comment by sbatten** [comment]
>bug seems to be different, not the position of the menu icon but the context menu that pops

Yes, that is what the bug is filed against, and I can do the work to fix that, but it won't be necessary if the fix is to move the menu back into the title bar since this only occurs because the compact menu is living in the the "top" version of the activity bar.

**Comment by sandy081** [comment]
> the fix is to move the menu back into the title bar since this only occurs because the compact menu is living in the the "top" version of the activity bar.

This will ask activity bar to also move to the title bar which makes it in-consistent across our products. I prefer to have compact menu action in the activity bar.

**Comment by arookas** [comment]
> @arookas Wondering how the same behaves when activity bar is on the side but hidden?

I tried this case (setting `workbench.activityBar.location` to `side` and going into Zen Mode). When I press Alt+F, no menus open. When I exit Zen Mode, the application menu appears messed up as in the first screenshot in OP.

> I think the right fix here is to open the sidebar when alt+f is pressed

I think this is fine; as long as Alt+ shortcuts function somehow.

> > the fix is to move the menu back into the title bar since this only occurs because the compact menu is living in the the "top" version of the activity bar.
>
> This will ask activity bar to also move to the title bar which makes it in-consistent across our products. I prefer to have compact menu action in the activity bar.

That's a shame. There's no way to move only the hamburger menu to the top left title bar and not the rest? The profile and settings buttons already do this in the top right title bar.

**Comment by sbatten** [comment]
@arookas you can always set `window.menubarvisibility` to `toggle` which will show it in the title bar hidden, activated on `alt` or `alt+<hotkey>` as a workaround.

@sandy081 
>This will ask activity bar to also move to the title bar which makes it in-consistent across our products. I prefer to have compact menu action in the activity bar.

I am not sure I follow. The activity bar stays in the side bar, it just doesn't show the compact menu, it relies on the titlebar to do so.

**Comment by sandy081** [comment]
I am fine moving the compact menu to the title bar, but this will trigger an ask to move activity bar actions also to the title bar.

<img width="444" alt="image" src="https://github.com/microsoft/vscode/assets/10746682/dc5b9c74-33c0-4d5b-8943-b84a176c7a01">

**Comment by arookas** [comment]
> @arookas you can always set `window.menubarvisibility` to `toggle` which will show it in the title bar hidden, activated on `alt` or `alt+<hotkey>` as a workaround.

I can confirm this works for me in my use case, thanks!

**Comment by sbatten** [comment]
@sandy081 I'm not sure why that ask would be triggered by this. We already are moving icons that previously existed to the title bar. The icons that have moved do not affect the current view container of the side bar. This is also true of the menu. But, it would not be true of the activity icons.

**Comment by sandy081** [comment]
I am fine with this change. Please feel free to add this.
